cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -985,6 +986,54 @@ TEST_F(TestS3FS, FileSystemFromUri) {
   AssertFileInfo(fs.get(), path, FileType::File, 8);
 }
 
+// Simple retry strategy that records errors encountered and its emitted retry delays
+class TestRetryStrategy : public S3RetryStrategy {
+ public:
+  virtual ~TestRetryStrategy() = default;
```

Comment:
As I wrote, the virtual destructor should probably be defined in the base class.